### DISCUSSION TITLE
Load image from dll ressource file

### DISCUSSION THREAD

**Discussion by Daoudou** [openscenegraph/OpenSceneGraph#1328] [Q&A]
hello 

I want to know if it's possible to load a image png from a .dll with OpenSceneGraph ?
i load the png with the function LoadImage (yes i on MFC) but i don't know how to give it to OpenSceneGraph. 

Thanks in advance

Daoudou

**Comment by robertosfield** [comment]
Try getting a pointer to the memory block then adapting this to a std::istream, then passing this to the to the appropriate RenderWriter::read(stream) method.  Sorry it's a bit high level, I've not used the OSG in any significant capacity since 2018.  For modern application development you are very likely better off using the OSG's successor the VSG.

**Reply by Daoudou** [reply]
Thanks 

i tried to do this , i get the memory block to a std::istream. but it a little difficulte to passing to the appropriate RenderWriter::read(stream) ^^'

**Reply by Daoudou** [reply]
IT'S A SUCCESSSSSSS

i write it here if sommeone want to do the same thing : 

` 
      HINSTANCE module = LoadLibrary(L"ressources3D.dll");
        if (module == NULL)
        {
        cout << "Load of the dll, failed" << endl;
        }
        else 
        {
        cout << "Load of the dll, success" << endl;

        HRSRC rsrc = FindResource(module, MAKEINTRESOURCE(IDB_PNG_TEST), L"PNG");
        if (!rsrc)
        {
            cout << "Load of the img from the dll, failed" << endl;
        }
        else
        {
            cout << "Load of the img from the dll, success" << endl;
            unsigned int myResourceSize = ::SizeofResource(module, rsrc);
            HGLOBAL myResourceData = ::LoadResource(module, rsrc);

            void* memoryOfPng = ::LockResource(myResourceData);
            size_t memory_file = SizeofResource(module, rsrc);
            string casted_memory(static_cast<char*>(memoryOfPng), memory_file);
            istringstream stream(casted_memory);

            osgDB::ReaderWriter* rw = osgDB::Registry::instance()->getReaderWriterForExtension("png");
            osgDB::ReaderWriter::ReadResult rr = rw->readImage(stream);
            if (rr.success())
                imgTest = rr.takeImage();
            
            cout << "test" << endl;
        }
    }`
